lib/proxy/http_download.rb

```diff
@@ -4,10 +4,14 @@ module Proxy
   class HttpDownload < Proxy::Util::CommandTask
     include Util
 
-    def initialize(src, dst)
+    def initialize(src, dst, options = {})
       @dst = dst
       wget = which("wget")
-      super("#{wget} --timeout=10 --tries=3 --no-check-certificate -nv -c \"#{escape_for_shell(src.to_s)}\" -O \"#{escape_for_shell(dst.to_s)}\"")
+      read_timeout = options[:read_timeout] || 10
```

Comment:
It's only three more parameters, probably no need for using a hash, esp. when the other two parameters are explicit.

Reply:
I had it as expicit first and then thought that it might not be the brightest  idea to be reliant on the ordering of the input params. For example in my tests I switched the order of  read_timeout and connect_timeout without realizing...

Reply:
Please either use explicit parameters here, or switch to named parameters (although I don't know if this has implications downstream, as these aren't available on rubies < 2.0).